addons/cloudfiles/views.py

```diff
@@ -0,0 +1,168 @@
+# -*- coding: utf-8 -*-
+# """Views fo the node settings page."""
+import httplib
+from flask import request
+import logging
+
+from django.core.exceptions import ValidationError
+import urllib
+
+from addons.cloudfiles.serializer import CloudFilesSerializer
+from addons.base import generic_views
+from rackspace import connection
+from openstack.exceptions import HttpException
+from framework.exceptions import HTTPError
+from framework.auth.decorators import must_be_logged_in
+
+from osf.models import ExternalAccount
+
+from website.project.decorators import (
+    must_have_addon,
+    must_have_permission,
+    must_be_addon_authorizer,
+)
+
+
+logger = logging.getLogger(__name__)
+debug = logger.debug
+
+SHORT_NAME = 'cloudfiles'
+FULL_NAME = 'Cloud Files'
+
+cloudfiles_account_list = generic_views.account_list(
+    SHORT_NAME,
+    CloudFilesSerializer
+)
+
+cloudfiles_import_auth = generic_views.import_auth(
+    SHORT_NAME,
+    CloudFilesSerializer
+)
+
+
+@must_be_logged_in
+def cloudfiles_add_user_account(auth, **kwargs):
+    """Verifies new external account credentials and adds to user's list"""
+    try:
+        secret_key = request.json['secretKey']
+        username = request.json['username']
+    except KeyError:
+        raise HTTPError(httplib.BAD_REQUEST)
+
+    if not (secret_key and username):
+        return {
+            'message': 'All the fields above are required.'
+        }, httplib.BAD_REQUEST
+
+    try:
+        conn = connection.Connection(username=username,
+                               api_key=secret_key,
+                               region='IAD')
```

Comment:
Is it valid to check against a hard-coded region? Is there a way to retrieve a valid region instead?